MatrixSDKTests/MXBackgroundSyncServiceTests.swift

```diff
@@ -0,0 +1,649 @@
+                    
+                    self.bgSyncService?.event(withEventId: eventId, inRoom: roomId) { (response) in
+                        switch response {
+                        case .success(let event):
+                            
+                            // -> The message can be read and decypted from MXBackgroundSyncService
+                            XCTAssertTrue(event.isEncrypted, "Event should be encrypted")
+                            XCTAssertNotNil(event.clear, "Event should be decrypted successfully")
+                            
+                            let text = event.content["body"] as? String
+                            XCTAssertEqual(text, Constants.messageText, "Event content should match")
+                            
+                            XCTAssertNil(bobStore.event(withEventId: eventId, inRoom: roomId), "Event should not be in session store yet")
+                            
+                            let syncResponseStore = MXSyncResponseFileStore()
+                            syncResponseStore.open(withCredentials: bobCredentials)
+                            XCTAssertNotNil(syncResponseStore.event(withEventId: eventId, inRoom: roomId), "Event should be stored in sync response store")
+                            
+                            // - Bob restarts their MXSession
+                            let newBobSession = MXSession(matrixRestClient: MXRestClient(credentials: bobCredentials, unrecognizedCertificateHandler: nil))
+                            newBobSession?.setStore(bobStore, completion: { (_) in
+                                newBobSession?.start(withSyncFilterId: bobStore.syncFilterId, completion: { (_) in
+                                    
+                                    // -> The message is available from MXSession and no more from MXBackgroundSyncService
+                                    XCTAssertNil(syncResponseStore.event(withEventId: eventId, inRoom: roomId), "Event should not be stored in sync response store anymore")
+                                    XCTAssertNotNil(bobStore.event(withEventId: eventId, inRoom: roomId), "Event should be in session store now")
+                                    expectation?.fulfill()
+                                })
+                            })
+                        case .failure(let error):
+                            XCTFail("Cannot fetch the event from background sync service - error: \(error)")
+                            expectation?.fulfill()
+                        }
+                    }
+                    
+                    break
+                case .failure(let error):
+                    XCTFail("Cannot set up initial test conditions - error: \(error)")
+                    expectation?.fulfill()
+                }
+            }
+        }
+        
+    }
+    
+    func testWithEncryptedEventRollingKeys() {
```

Comment:
Can you describe this test and next one as `testWithPlainEvent`?
The goal is to understand what they are testings when we will come to them in future.